### PR TITLE
try to make the output of build-order show-duplicates more readable

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

File: src/fromager/commands/build_order.py [hunk 1]
```diff
@@ -298,12 +298,22 @@ def show_duplicates(build_order_file: pathlib.Path, install_only: bool):
...
+            print(f"  {req}")
+            if len(why) > 1:
+                parents = " -- ".join(f"'{w[1]}'" for w in why[:-1])
+                print(
+                    f"    resolved to {step['version']} as {article} {req_type} dependency of {parents}"
+                )
+            else:
+                print(
+                    f"    resolved to {step['version']} as {article} {req_type} dependency"
+                )
```
Comment (shubhbapna) [inline src/fromager/commands/build_order.py:319]:
```suggestion
            parents = " of " + " -- ".join(f"'{w[1]}'" for w in why[:-1]) if len(why) > 1 else ""
            print(f"\t resolved to {step['version']} as {article} {req_type} dependency{parents}")
```

**Review by shubhbapna — COMMENTED** [review]
Instead of white space a tab character might be more consistent?

File: src/fromager/commands/build_order.py (hunk 1 above)
Comment (dhellmann) [reply src/fromager/commands/build_order.py:319]: Do you find that easier to read? It seems more complex to me, especially with all of the quotes.

Comment (shubhbapna) [reply src/fromager/commands/build_order.py:319]: fair i was trying to reduce the duplicate print statements. How about something like this?

```python
show_duplicate_msg = f"\t resolved to {step['version']} as {article} {req_type} dependency"
if len(why) > 1:
    parents = " -- ".join(f"'{w[1]}'" for w in why[:-1])
    show_duplicate_msg = f"{show_duplicate_msg} of {parents}"
print(show_duplicate_msg)
```

Comment (dhellmann) [reply src/fromager/commands/build_order.py:319]: That looks better, I'll make that update when I come back to this.

Comment (dhellmann) [reply src/fromager/commands/build_order.py]: Updated

**Pull request merged**